app/src/main/java/pl/example/android/garbageapp/data/SectorTermRepository.java

```diff
@@ -61,13 +63,30 @@ public synchronized static SectorTermRepository getInstance(SectorTermDao sector
     }
 
     private synchronized void initializeData() {
-        mExecutors.diskIO().execute(() -> startFetchSectorTerms());
+        if (mInitialized) {
+            return;
+        }
+        mInitialized = true;
+
+        mSectorTermsNetworkDataSource.scheduleRecurringFetchSectorTermsSync();
+
+        mExecutors.diskIO().execute(() -> {
+            if (isFetchNeeded()) {
+                startFetchSectorTerms();
+            }
+        });
     }
 
     private void startFetchSectorTerms() {
         mSectorTermsNetworkDataSource.startSectorTermsSyncService();
     }
 
+    private boolean isFetchNeeded() {
+        Date today = SectorTermsDateUtils.getNormalizedUtcDateForToday();
+        int futureWeather = mSectorTermDao.countAllFutureSectorTerms(today);
+        return futureWeather < SectorTermsNetworkDataSource.NUM_DAYS;
```

Comment:
This condition is wrong. Dao countAllFutureSectorTerms will return number of terms. We cannot compare it to number of days.